### ISSUE TITLE
Unable to assert invalid dates

### ISSUE THREAD

**Issue by meandmycode** [mochajs/mocha#1660]
When asserting something that may return an invalid date mocha will error trying to report the invalid date due to calling ```toISOString ``` on an invalid date.

Example (using should):

```js
describe('mocha', function() {
  it('should not error when comparing invalid dates', function() {
    new Date('').should.equal(new Date(''));
  })
})
```

I can submit a PR if it helps, I think the stringify utility just needs to call ```toString()``` in the event that the date is invalid, which will result in "[Date: Invalid Date]"

**Comment by a8m** [comment]
Thanks @meandmycode 
PTAL on #1661

**Comment by meandmycode** [comment]
Much appreciated, speedy work!

**Comment by boneskull** [comment]
closed via #1661
